src/core/src/bootstrap/ConfigurationFactory.py

```diff
@@ -240,4 +255,36 @@ def new_test_configuration(self, package_manager_name, package_manager_component
         configuration['config_env'] = Constants.TEST
         # perform desired modifications to configuration
         return configuration
+
+    def get_lifecycle_manager_component(self, vm_cloud_type):
+        """ finding life cycle manager based on vm and returning component name added in the prod configuration """
+        azure_lifecycle_manager_component = LifecycleManagerAzure
+        arc_lifecycle_manager_component = LifecycleManagerArc
+        if(vm_cloud_type == Constants.VMCloudType.AZURE):
+            return azure_lifecycle_manager_component
+        elif (vm_cloud_type == Constants.VMCloudType.ARC):
+            return arc_lifecycle_manager_component
+        
+        return azure_lifecycle_manager_component
+   
+    def get_vm_cloud_type(self):
+        """ detects vm type. logic taken from HCRP code: https://github.com/PowerShell/DesiredStateConfiguration/blob/dev/src/dsc/dsc_service/service_main.cpp#L115
+        Todo:  how to check this only when it is Auto Assessment operation??? """
```

Comment:
Given the thread about how IMDS is unreliable and that this will touch production Azure, until PR for TODO in 272 happens, please use a retry policy similar to write_with_retry in EnvLayer.

Also use print statements to get code behavior here instrumented and logged into the log file.

Please plan on working on the operation detection immediately after this PR as this is the most worrisome part of this code.